velox/type/Type.h

```diff
@@ -991,58 +1085,62 @@ std::shared_ptr<const OpaqueType> OPAQUE() {
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::SMALLINT: {                      \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::SMALLINT>(  \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::BIGINT: {                        \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::BIGINT>(    \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::REAL: {                          \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::REAL>(      \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::DOUBLE: {                        \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::DOUBLE>(    \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::VARCHAR: {                       \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::VARCHAR>(   \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::VARBINARY: {                     \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::VARBINARY>( \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::TIMESTAMP: {                     \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::TIMESTAMP>( \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      case ::facebook::velox::TypeKind::DATE: {                          \
-        return TEMPLATE_FUNC<T, ::facebook::velox::TypeKind::DATE>(      \
-            __VA_ARGS__);                                                \
-      }                                                                  \
-      default:                                                           \
-        VELOX_FAIL(                                                      \
-            "not a scalar type! kind: {}", mapTypeKindToName(typeKind)); \
-    }                                                                    \
+#define VELOX_DYNAMIC_SCALAR_TEMPLATE_TYPE_DISPATCH(                         \
```

Comment:
Nit : Why is this diff showing in entirety vs just showing the lines changed for the DECIMAL types ?

Reply:
May be the use of `\` treats it like a single statement?